extending-valtimo/plugin/custom-plugin-definition.md

```diff
@@ -70,3 +72,448 @@ class TwitterPluginFactory(
     }
 }
 ```
+
+## Front-end
+
+To develop your own front-end plugin, the library `@valtimo/plugin` provides several interfaces which your own front-end
+plugin must conform to, in order to be used in your implementation.
+
+### Your plugin specification
+
+First you need to make a plugin specification, conforming to the interface `PluginSpecification`. Below is an example
```

Comment:
First, a plugin specification conforming to the `PluginSpecification` interface needs to be created.